addons/mail/models/res_partner.py

```diff
@@ -234,10 +234,8 @@ def mail_partner_format(self, fields=None):
             if 'user' in fields:
                 internal_users = partner.user_ids - partner.user_ids.filtered('share')
                 main_user = internal_users[0] if len(internal_users) > 0 else partner.user_ids[0] if len(partner.user_ids) > 0 else self.env['res.users']
-                data['user'] = {
-                    "id": main_user.id,
-                    "isInternalUser": not main_user.share,
-                } if main_user else False
+                data['userId'] = main_user.id if main_user else False
+                data["isInternalUser"] = not main_user.share if main_user else False
```

Comment:
fallback to False is automated with python recordset/fields, there is no need for if/else case